modules/core/src/lib/layer.js

```diff
@@ -59,7 +59,28 @@ const defaultProps = {
   onDataLoad: {type: 'function', value: null, compare: false, optional: true},
   fetch: {
     type: 'function',
-    value: (url, {layer}) => load(url, layer.getLoadOptions()),
+    value: (url, {propName, layer}) => {
+      const {dataManager} = layer.context;
+      const loadOptions = layer.getLoadOptions();
+
+      if (dataManager) {
+        if (!loadOptions && !dataManager.contains(url)) {
+          // If there is no layer-specific load options, then attempt to cache this resource in the data manager
+          dataManager.add(url, url, {persistent: false});
+        }
+        const resource = dataManager.subscribe(
+          url,
+          data => layer.internalState.reloadAsyncProp(propName, data),
+          layer.id,
+          propName
+        );
+        if (resource) {
+          return resource;
```

Comment:
What is the type of a resource? Isn't that a wrapper? Don't we want to return some data value or promise?